custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -780,25 +784,80 @@ def _prepare_prediction_server_or_batch_pipeline(self, run_language):
 
         return DrumUtils.render_file(functional_pipeline_filepath, replace_data)
 
-    def _run_predictions_pipelines_in_mlpiper(self):
-        run_language = self._check_artifacts_and_get_run_language()
+    def _run_predictions(self):
+        if self.run_mode != RunMode.SCORE:
+            raise NotImplemented("Only score mode is supported here")
 
-        if self.run_mode == RunMode.SERVER:
-            # in prediction server mode infra pipeline == prediction server runner pipeline
-            infra_pipeline_str = self._prepare_prediction_server_or_batch_pipeline(run_language)
-        elif self.run_mode == RunMode.SCORE:
-            tmp_output_filename = None
-            # if output is not provided, output into tmp file and print
-            if not self.options.output:
-                # keep object reference so it will be destroyed only in the end of the process
-                __tmp_output_file = tempfile.NamedTemporaryFile(mode="w")
-                self.options.output = tmp_output_filename = __tmp_output_file.name
-            # in batch prediction mode infra pipeline == predictor pipeline
+        with self._setup_output_if_not_exists():
+            run_language = self._check_artifacts_and_get_run_language()
             infra_pipeline_str = self._prepare_prediction_server_or_batch_pipeline(run_language)
+
+            pipeline = json.loads(infra_pipeline_str)
+            if "pipe" not in pipeline or not pipeline["pipe"]:
+                raise DrumCommonException("Pipeline is empty")
+            if "arguments" not in pipeline["pipe"][0]:
```

Comment:
Should `"pipe"` and `"arguments"` be enums?

Reply:
I would do it, however, a follow up project is supposed to cleanup all of these code. This `pipeline` is not longer needed and will be removed completely.